anyblok/plugins.py

```diff
@@ -80,66 +87,56 @@ def configure(self, options, conf):
-            if registry:
-                installed_bloks = registry.System.Blok.list_by_state(
-                    "installed")
-                selected_bloks = return_list(
-                    Configuration.get('selected_bloks')) or installed_bloks
-
-                unwanted_bloks = return_list(
-                    Configuration.get('unwanted_bloks')) or []
-
-                self.bloks_path = [BlokManager.getPath(x)
-                                   for x in BlokManager.ordered_bloks]
-
-                self.authoried_bloks_test_files = []
-                for blok in installed_bloks:
-                    if blok not in selected_bloks or blok in unwanted_bloks:
-                        continue
-
-                    startpath = BlokManager.getPath(blok)
-                    for root, dirs, _ in walk(startpath):
-                        if 'tests' in dirs:
-                            self.authoried_bloks_test_files.append(
-                                join(root, 'tests'))
-
-                registry.close()  # free the registry to force create it again
-
-    def file_from_blok(self, file):
-        for blok_path in self.bloks_path:
-            if file.startswith(blok_path):
-                return True
+        if not self.enabled or self.registryLoaded:
+            return
 
-        return False
+        # Load the registry here not in configuration,
+        # because the configurations are not loaded in order of score
+        self.registryLoaded = True
+        load_init_function_from_entry_points(unittest=True)
+        Configuration.load_config_for_test()
+        Configuration.parse_options(self.AnyBlokOptions)
+        configuration_post_load()
+        BlokManager.load()
+        db_name = get_db_name()
 
-    def file_from_authorized_bloks(self, file):
-        for testFile in self.authoried_bloks_test_files:
-            if file.startswith(testFile):
-                return True
+        registry = RegistryManager.get(db_name)
+        if not registry:
+            return
 
-        return False
+        installed_bloks = registry.System.Blok.list_by_state("installed")
+        selected_bloks = return_list(
+            Configuration.get('selected_bloks')) or installed_bloks
+
+        unwanted_bloks = return_list(Configuration.get('unwanted_bloks')) or []
+
+        self.authorized_blok_paths = set(
+            BlokManager.getPath(b) for b in BlokManager.list()
+            if b in selected_bloks and b not in unwanted_bloks)
+
+        test_dirs = self.authorized_blok_test_dirs = set()
+        for startpath in self.authorized_blok_paths:
+            for root, dirs, _ in walk(startpath):
+                if 'tests' in dirs:
+                    test_dirs.add(join(root, 'tests'))
+
+        registry.close()  # free the registry to force create it again
+
+    def file_from_authorized_blok_tests(self, file_path):
+        return any(isindir(file_path, tp)
+                   for tp in self.authorized_blok_test_dirs)
```

Comment:
In particular, this is not a simple `startswith` anymore, it's actually testing we are in the directory. That should prevent a few subtle bugs (assuming we keep this restriction to "tests" subdirectories)